clr-k8s-examples/setup_system.sh

```diff
@@ -3,85 +3,159 @@
+  sudo systemctl enable --now kubelet $RUNNER || true
+}
 
-#Ensure that the system is ready without requiring a reboot
-sudo swapoff -a
-sudo systemctl restart systemd-modules-load.service
+# ensure that the system is ready without requiring a reboot
+function ensure_system_ready() {
+  sudo swapoff -a
+  sudo systemctl restart systemd-modules-load.service
+}
 
+# add proxy if found
+function setup_proxy(){
 set +o nounset
 if [[ ${http_proxy} ]] || [[ ${HTTP_PROXY} ]]; then
-	echo "Setting up proxy stuff...."
-	# Setup IP for users too
-	sed_val=${ADD_NO_PROXY//\//\\/}
-	[ -f /etc/environment ] && sudo sed -i "/no_proxy/I s/$/,${sed_val}/g" /etc/environment
-	if [ -f /etc/profile.d/proxy.sh ]; then
-		sudo sed -i "/no_proxy/I s/\"$/,${sed_val}\"/g" /etc/profile.d/proxy.sh
-	else
-		echo "Warning, failed to find /etc/profile.d/proxy.sh to edit no_proxy line"
-	fi
-
-	services=($RUNNER 'kubelet')
-	for s in "${services[@]}"; do
-		sudo mkdir -p "/etc/systemd/system/${s}.service.d/"
-		cat <<EOF | sudo bash -c "cat > /etc/systemd/system/${s}.service.d/proxy.conf"
+  echo "Setting up proxy stuff...."
+  # Setup IP for users too
+  sed_val=${ADD_NO_PROXY//\//\\/}
+  [ -f /etc/environment ] && sudo sed -i "/no_proxy/I s/$/,${sed_val}/g" /etc/environment
+  if [ -f /etc/profile.d/proxy.sh ]; then
+    sudo sed -i "/no_proxy/I s/\"$/,${sed_val}\"/g" /etc/profile.d/proxy.sh
+  else
+    echo "Warning, failed to find /etc/profile.d/proxy.sh to edit no_proxy line"
+  fi
+
+  services=($RUNNER 'kubelet')
+  for s in "${services[@]}"; do
+    sudo mkdir -p "/etc/systemd/system/${s}.service.d/"
+    cat <<EOF | sudo bash -c "cat > /etc/systemd/system/${s}.service.d/proxy.conf"
 [Service]
 Environment="HTTP_PROXY=${http_proxy}"
 Environment="HTTPS_PROXY=${https_proxy}"
 Environment="SOCKS_PROXY=${socks_proxy}"
 Environment="NO_PROXY=${no_proxy},${ADD_NO_PROXY}"
 EOF
-	done
+  done
 fi
 set -o nounset
+}
+
+# init for performing any pre tasks
+function init(){
+  echo ""
+}
+
+###
+# main
+##
+set +o nounset
+if [[ -n "${CLRK8S_OS}" ]]; then
+  # shellcheck disable=SC1090
+  source "$(dirname "$0")/setup_system_${CLRK8S_OS}.sh"
+fi
+set -o nounset
+
+echo "Init..."
+init
+echo "Setting OS Version..."
+upate_os_version
+echo "Adding OS Dependencies..."
+add_os_deps
+echo "Disabling swap..."
+disable_swap
+echo "Enabling IP Forwarding..."
+enable_ip_forwarding
+echo "Setting up modules to load..."
+setup_modules_load
+echo "Setting up /etc/hosts..."
+setup_hosts
+echo "Reloading daemons..."
+daemon_reload
+echo "Enabling Kublet runner..."
+enable_kubelet_runner
+echo "Ensuring system is ready..."
+ensure_system_ready
+echo "Detecting and setting up proxy..."
+setup_proxy
 
 # We have potentially modified their env files, we need to restart the services.
+# daemon reload
 sudo systemctl daemon-reload
+# restart runner
 sudo systemctl restart $RUNNER || true
+# restart kubelet
+sudo rm /var/lib/kubelet/cpu_manager_state
```

Comment:
Thought kubeadm reset takes care of this?

Reply:
I was running into issue but will take it out.

Reply:
I moved this to the setup_system_ubuntu.sh (not in tree) so it has been removed.